### PR TITLE
Define cloudwatch rule and SQS queue

### PR DESCRIPTION
**Pull request by xbrianh**
This defines infrastructure that will be used for the forthcoming refactor of the DSS event journaling and update daemon:
- A scheduled CloudWatch event rule
- An SQS queue
- CloudWatch-SQS integration

connected to #2648

### CODE REVIEW

File: infra/dss-events-scribe/main.tf [hunk 1]
```diff
@@ -0,0 +1,76 @@
+data aws_caller_identity current {}
+data aws_region current {}
+
+locals {
+  region = "${data.aws_region.current.name}"
+  account_id = "${data.aws_caller_identity.current.account_id}"
+  common_tags = "${map(
+    "managedBy" , "terraform",
+    "Name"      , "${var.DSS_INFRA_TAG_SERVICE}-subscriptionsdynamodb",
```
Comment (amarjandu) [inline infra/dss-events-scribe/main.tf]: please change the name

File: infra/dss-events-scribe/main.tf [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:lambda:${local.region}:${local.account_id}:function:dss-events-scribe-${var.DSS_DEPLOYMENT_STAGE}"]
+    }
+  }
+}
+
+resource "aws_sqs_queue" "dss-events-scribe-queue" {
+  name   					 = "dss-events-scribe-${var.DSS_DEPLOYMENT_STAGE}"
+  tags   					 = "${local.common_tags}"
+  message_retention_seconds  = "3600"
```
Comment (DailyDreaming) [inline infra/dss-events-scribe/main.tf]: What happens to the message after an hour?

File: infra/dss-events-scribe/main.tf [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  tags 				  = "${local.common_tags}"
+}
+
+resource "aws_cloudwatch_event_target" "send-journal-and-update-message" {
+  count = "${length(local.replicas)}"
+  rule  = "${aws_cloudwatch_event_rule.events-scribe.name}"
+  arn   = "${aws_sqs_queue.dss-events-scribe-queue.arn}"
+  input = <<-DOC
+  {
+    "replica":"${local.replicas[count.index]}"
```
Comment (amarjandu) [inline infra/dss-events-scribe/main.tf]: should this key be 'replicas' ?

File: infra/dss-events-scribe/main.tf (hunk 2 above)
Comment (xbrianh) [reply infra/dss-events-scribe/main.tf]: They're discarded.

File: infra/dss-events-scribe/main.tf (hunk 1 above)
Comment (xbrianh) [reply infra/dss-events-scribe/main.tf]: will fix

File: infra/dss-events-scribe/main.tf (hunk 3 above)
Comment (xbrianh) [reply infra/dss-events-scribe/main.tf]: Nope. This block defines two CloudWatch event targets, one for each replica, so the input is either `{"replica": "aws"}` or `{"replica": "gcp"}`

File: infra/dss-events-scribe/main.tf (hunk 2 above)
Comment (chmreid) [reply infra/dss-events-scribe/main.tf]: I believe that is answered by the documentation added in #2656: https://github.com/HumanCellAtlas/data-store/pull/2656/files#diff-431e856038c071e599fc68939e8d4365R18

> Messages older than `M > N` minutes are discarded from the queue

File: infra/dss-events-scribe/main.tf [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    principals {
+      type = "AWS"
+      identifiers = ["*"]
+    }
+    actions = ["sqs:SendMessage"]
+    resources = ["arn:aws:sqs:${local.region}:${local.account_id}:dss-events-scribe-${var.DSS_DEPLOYMENT_STAGE}"]
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+      values = ["${aws_cloudwatch_event_rule.events-scribe.arn}"]
```
Comment (chmreid) [inline infra/dss-events-scribe/main.tf]: Just to make sure I understand correctly - this policy gives any cloudwatch event rule-based principal permission to send a message to the SQS queue used for event journaling?

File: infra/dss-events-scribe/main.tf [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      identifiers = ["*"]
+    }
+    actions = [
+      "sqs:*",
+    ]
+    resources = ["arn:aws:sqs:${local.region}:${local.account_id}:dss-events-scribe-${var.DSS_DEPLOYMENT_STAGE}"]
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:lambda:${local.region}:${local.account_id}:function:dss-events-scribe-${var.DSS_DEPLOYMENT_STAGE}"]
```
Comment (chmreid) [inline infra/dss-events-scribe/main.tf]: And likewise, this policy gives the dss-events-scribe lambda function permission to take any action on the event journaling SQS queue?

File: infra/dss-events-scribe/main.tf (hunk 1 above)
Comment (chmreid) [inline infra/dss-events-scribe/main.tf]: is this `subscriptionsdynamodb` an artifact from copying and pasting?

Comment (chmreid) [reply infra/dss-events-scribe/main.tf]: nm, already mentioned by @amarjandu

File: infra/dss-events-scribe/main.tf (hunk 4 above)
Comment (xbrianh) [reply infra/dss-events-scribe/main.tf]: yep

File: infra/dss-events-scribe/main.tf (hunk 5 above)
Comment (xbrianh) [reply infra/dss-events-scribe/main.tf]: Yes, that's correct, although the lambda execution role defined [here](https://github.com/HumanCellAtlas/data-store/blob/master/iam/policy-templates/dss-events-scribe-lambda.json) can further restrict the SQS actions the available to dss-events-scribe

**Pull request merged**